### DISCUSSION TITLE
Boot time

### DISCUSSION THREAD

**Discussion by erfanium** [nanovms/ops#1217] [Q&A]
Hi, Thank you for creating this amazing tool!

I created a hello-world http server app for nodejs and run it with ‍‍`ops`. it's works but the overall boot time is near **3 seconds** for me.  

Two questions
1. Is that normal? due to claims that have been made, I expected a boot time less than a second.

2. Should I use `XEN` or `KVM` instead of default `QEMU` to boost boot time? If yes, how can i do that? I can't find anything related to `KVM` or `XEN` in the docs.
 
my host OS: ubuntu 18.04 64bit
CPU: Intel core i5 7400
Using the default QEMU as hypervisor

**Comment by francescolavra** [comment]
Most of the nearly 3 seconds you are measuring is likely due to the Qemu startup time. In general, the boot time you should expect for a given application is roughly the sum of a hypervisor-specific startup time and the application startup time; the application startup time should be comparable with the time you need to launch the same application under Linux.
Qemu uses KVM (on Linux) or HVF (on macOS), so when you use `ops run` you are already using KVM (unless hardware acceleration is disabled, in which case you would see a warning when you run Ops). XEN is currently not supported when running instances on premises. If you want fast boot time you could look at using Firecracker: if I run the node.js web server at https://github.com/nanovms/nanos/blob/master/test/e2e/node_v11.5.0/hi.js with Nanos on Firecracker, it takes less than one second for the 'Server running at http...' message to show up in the terminal. For more info on using Firecracker, see https://docs.ops.city/ops/firecracker.

**Reply by erfanium** [reply]
> Qemu uses KVM (on Linux) or HVF (on macOS), so when you use ops run you are already using KVM

I know it, but QEMU is a type 2 hypervisor so it's pretty slow in boot time and other performance metrics, maybe it's better to use `KVM` directly. (KVM itself is a type 1 hypervisor)

Thank you for recommending Firecracker, I'll test it in the next week
